src/common/ownsql.cpp

```diff
@@ -17,26 +17,35 @@
  */
 
 #include <QDateTime>
-#include <QLoggingCategory>
-#include <QString>
+#include <QDir>
 #include <QFile>
 #include <QFileInfo>
-#include <QDir>
+#include <QLoggingCategory>
+#include <QString>
 
-#include "ownsql.h"
-#include "common/utility.h"
 #include "common/asserts.h"
+#include "common/utility.h"
+#include "ownsql.h"
+
 #include <sqlite3.h>
 
-#define SQLITE_SLEEP_TIME_USEC 100000
-#define SQLITE_REPEAT_COUNT 20
+#include <chrono>
+#include <thread>
+
+using namespace std::chrono_literals;
 
-#define SQLITE_DO(A)                                         \
-    if (1) {                                                 \
-        _errId = (A);                                        \
-        if (_errId != SQLITE_OK && _errId != SQLITE_DONE && _errId != SQLITE_ROW) {  \
-            _error = QString::fromUtf8(sqlite3_errmsg(_db)); \
-        }                                                    \
+namespace {
+constexpr auto SQLITE_SLEEP_TIME = 500ms;
```

Comment:
The sleep is probably not needed as it seems that the sqlite calls them self do block for a certain amount of time.